src/OneDriveSdk/Requests/BaseClient.cs

```diff
@@ -71,7 +71,7 @@ public IAuthenticationProvider AuthenticationProvider
         public string BaseUrl
         {
             get { return this.baseUrl; }
-            set { this.baseUrl = value.TrimEnd('/'); }
+            set { this.baseUrl = string.IsNullOrEmpty(value) ? null : value.TrimEnd('/'); }
```

Comment:
Is this for ReSharper?  If you assign null, empty, whitespace to BaseUrl I would expect an InvalidArgument exception